conans/test/functional/toolchains/gnu/test_makedeps.py

```diff
@@ -0,0 +1,119 @@
+import textwrap
+
+import platform
+import pytest
+
+from conans.test.assets.autotools import gen_makefile
+from conans.test.assets.sources import gen_function_cpp, gen_function_h
+from conans.test.utils.tools import TestClient
+from conan.tools.gnu.makedeps import CONAN_MAKEFILE_FILENAME
+
+
+@pytest.mark.tool("make" if platform.system() != "Windows" else "msys2")
+def test_make_deps_definitions_escape():
+    """
+    MakeDeps has to escape the definitions properly.
+    """
+    client = TestClient(path_with_spaces=False)
+    conanfile = textwrap.dedent(r'''
+        from conan import ConanFile
+        class HelloLib(ConanFile):
+            def package_info(self):
+                self.cpp_info.defines.append("USER_CONFIG=\"user_config.h\"")
+                self.cpp_info.defines.append('OTHER="other.h"')
+                self.cpp_info.cflags.append("flag1=\"my flag1\"")
+                self.cpp_info.cxxflags.append('flag2="my flag2"')
+        ''')
+    client.save({"conanfile.py": conanfile})
+    client.run("export . --name=hello --version=0.1.0")
+    client.run("install --requires=hello/0.1.0 --build=missing -g MakeDeps")
+    client.run_command(f"make --print-data-base -f {CONAN_MAKEFILE_FILENAME}", assert_error=True)
+    assert r'CONAN_CXXFLAGS_HELLO = flag2=\"my flag2\"' in client.out
+    assert r'CONAN_CFLAGS_HELLO = flag1=\"my flag1\"' in client.out
+    assert r'CONAN_DEFINES_HELLO = $(CONAN_DEFINE_FLAG)USER_CONFIG="user_config.h" $(CONAN_DEFINE_FLAG)OTHER="other.h"' in client.out
+
+
+@pytest.mark.tool("cmake")
+def test_makedeps_with_tool_requires():
```

Comment:
This is a very expensive test, with creation of 3 full packages, compiled libraries, for very light checking, that could be done with empty recipes that do not build anything. So this could be an integration test that doesn't depend on cmake at all

Reply:
Done. No longer using build step, but only checking cpp_info through dependencies.